tests/network/primary_pod_network.go

```diff
@@ -85,60 +76,111 @@ var _ = SIGDescribe("Primary Pod Network", func() {
 			var vmi *v1.VirtualMachineInstance
 
 			BeforeEach(func() {
-				vmi = setupVMI(virtClient, vmiWithMasqueradeBinding())
+				vmi = libvmi.SetupVMI(vmiWithMasqueradeBinding(), tests.LoggedInCirrosExpecter)
 			})
 
 			AfterEach(func() {
-				cleanupVMI(virtClient, vmi)
+				libvmi.CleanupVMI(vmi)
 			})
 
 			It("should report PodIP as its own on interface status", func() { AssertReportedIP(vmi) })
 		})
 	})
-})
 
-func setupVMI(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance) *v1.VirtualMachineInstance {
-	By("Creating the VMI")
-	var err error
-	vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
-	Expect(err).NotTo(HaveOccurred(), "VMI should be successfully created")
+	Describe("Connectivity", func() {
+		const servicePort = 1500
+		const ipv4HeaderSize = 20
+		const icmpHeaderSize = 8
+		const ipv4IcmpHeaderSize = ipv4HeaderSize + icmpHeaderSize
 
-	By("Waiting until the VMI gets ready")
-	vmi = tests.WaitUntilVMIReady(vmi, tests.LoggedInAlpineExpecter)
+		AssertConnectivity := func(outboundVMI, inboundVMI *v1.VirtualMachineInstance, checkIPv6 bool) {
+			for _, inboundVMIAddress := range inboundVMI.Status.Interfaces[0].IPs {
+				if net.IsIPv6String(inboundVMIAddress) && !checkIPv6 {
+					continue
+				}
 
-	return vmi
-}
+				By("Checking connectivity via ICMP")
+				Expect(tests.PingFromVMConsole(outboundVMI, inboundVMIAddress)).To(Succeed())
 
-func cleanupVMI(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance) {
-	if vmi != nil {
-		By("Deleting the VMI")
-		Expect(virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(vmi.GetName(), &metav1.DeleteOptions{})).To(Succeed())
-
-		By("Waiting for the VMI to be gone")
-		Eventually(func() error {
-			_, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmi.GetName(), &metav1.GetOptions{})
-			return err
-		}, 2*time.Minute, time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())), "The VMI should be gone within the given timeout")
-	}
-}
+				By("Checking connectivity via ICMP with MTU-sized frames")
+				mtu := getPodNetworkMTU(outboundVMI)
+				Expect(tests.PingFromVMConsole(outboundVMI, inboundVMIAddress, "-c5", "-w5", "-s", strconv.Itoa(mtu-ipv4IcmpHeaderSize))).To(Succeed())
```

Comment:
ipv4IcmpHeaderSize 
if we do check ipv6, dont we want ipv6 + icmp size here ?

Reply:
Great catch, thanks! Maybe that explains some of the failing tests. IPv6 header size should be 40 bytes.

Reply:
i think it wont fix those problems, (also saw some tests fail on the IPv4 in the lanes of this PR , 1.18)
i would even always reduce 40 + 8 or so, for simplicity, and easier to maintain.
the mtu is 1440, and here we just affect by few bytes the "-s SIZE", and doesn't have other affect.